framework/sessions/__init__.py

```diff
@@ -58,11 +81,28 @@ def prepare_private_key():
     if request.method != 'GET':
         return
 
-    # Done if private_key in args
+    node_id = get_node_id(request.url)
```

Comment:
Also, you don't need to do this unless you know there's a private link. Otherwise this will slow every request down.